vm/src/stdlib/io.rs

```diff
@@ -109,12 +109,35 @@ impl BufferedIO {
         self.cursor.position()
     }
 
-    fn readline(&mut self) -> Option<String> {
-        let mut buf = String::new();
+    fn readline(&mut self, size: OptionalOption<i64>, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
+        let mut buf = Vec::new();
+        let mut left = size.flat_option().filter(|s| *s >= 0).unwrap_or(i64::MAX);
+        if left == 0 {
+            return Ok(buf);
+        }
+        loop {
+            let (done, used) = {
+                let mut available = self.cursor.fill_buf().map_err(|err| os_err(vm, err))?;
+                if left < available.len() as i64 {
+                    available = &available[..left as usize];
```

Comment:
It seems i64 to usize casting is not safe for 32bit platforms

Reply:
I think this was safe as `available.len()` is a `usize`, so `left` can always fit into 32bits here. I have changed it so that `left` is now also a `usize`, which removes a number of casts.